newscoop/src/Newscoop/NewscoopBundle/Security/User/SimpleUserProvider.php

```diff
@@ -0,0 +1,73 @@
+<?php                                                                                                   
+namespace Newscoop\NewscoopBundle\Security\User;
+
+use Newscoop\Entity\User;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\Exception\UsernameNotFoundException;
+use Symfony\Component\Security\Core\Exception\UnsupportedUserException;
+
+class SimpleUserProvider implements UserProviderInterface
+{
+    protected $users;
+
+    protected $em;
+
+    protected $config;
+
+    public function __construct($em, array $config)
+    {
+        $this->em = $em;
+        $this->config = $config;
+        if (file_exists($this->config['users_file'])) {
+            $this->users = json_decode(file_get_contents($this->config['users_file']), true);
+        } else {
+            $this->users = array();
+        }
+    }
+
+    public function loadUserByUsername($username)
+    {
+        if (isset($this->users[$username])) {
+            // query User entity
+            $user = $this->em->getRepository('Newscoop\Entity\User')->findOneByUsername($username);
+
+            // create the user if they don't already exist
+            if (empty($user)) {
+                // create user
+                $user = new User();
+                $this->em->getRepository('Newscoop\Entity\User')->save($user, array(
+                    'username' => $username,
+                    'email' => $username,
+                    'password' => $this->users[$username],
```

Comment:
you mean $this->users[$username] ?  I thought would only not work if $this->users was a function (its just an array here).  Should I just do:

$users = $this->users;
...
'password' =>  $users[$username],
...